src/sugar3/activity/bundlebuilder.py

```diff
@@ -345,6 +411,61 @@ def _install_desktop_file(self, prefix, activity_path):
         with open(path, 'w') as f:
             cp.write(f)
 
+    def _generate_appdata(self, prefix, activity_path):
+        info = ConfigParser()
+        info.read(os.path.join(activity_path, 'activity', 'activity.info'))
+
+        required_fields = ['metadata_license', 'license', 'name', 'icon',
+                           'description']
+        for name in required_fields:
+            if not info.has_option('Activity', name):
+                print('[WARNING] Activity needs more metadata for AppStream '
+                      'file')
+                print('  Without an AppStream file, the activity will NOT '
+                      'show in software stores!')
+                print('  Please `pydoc sugar3.activity.bundlebuilder` for'
+                      'more info')
+                return
+
+        # See https://people.freedesktop.org/~hughsient/appdata/ for spec
```

Comment:
Please don't use the old location, and rather the "new" spec:
https://www.freedesktop.org/software/appstream/docs/sect-Metadata-Application.html, which documents all valid tags excessively, or the quickstart guide: https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html#sect-Quickstart-DesktopApps